pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -300,19 +317,75 @@ func (p *PriorityQueue) Run() {
 	go wait.Until(p.flushUnschedulablePodsLeftover, 30*time.Second, p.stop)
 }
 
+// PreEnqueue firstly runs registered EnqueuePlugins, and returns true if all plugins run
+// successfully; otherwise add or update the pod in `unschedulablePods`, and returns false.
+func (p *PriorityQueue) PreEnqueue(pInfo *framework.QueuedPodInfo) bool {
+	success := p.runEnqueuePlugins(context.Background(), pInfo)
+	if !success {
+		// Add the Pod to unschedulablePods if it's not passing EnqueuePlugins.
+		p.unschedulablePods.addOrUpdate(pInfo)
+	}
+	pInfo.Gated = !success
+	return success
```

Comment:
I find this function redundant, can we just move this logic into `addToActiveQ`, see the comment on that function.

Reply:
also, why are we exporting it?